Adding example projects (#151)
dfinity/examples: hosting/static-website/README.md

# Static website
The example shows how to deploy a simple, static website on the IC in an asset canister. The website is very simple, it just shows the Dfinity logo and has a basic styling of the logo. 

![Website](README_images/website.png)

## Introduction

> The purpuse of this example is to show how to deploy a static website in an asset canister. While the website in this example is very simple, the method is the same for more advanced static website, e.g. based on popular static site generators.

Reviewer: ```suggestion
The purpose of this example is to show how to deploy a static website in an asset canister. While the website in this example is very simple, the method is the same for more advanced static website, e.g. based on popular static site generators.
```

> The purpose of this example is to show how to deploy a static website in an asset canister. While the website in this example is very simple, the method is the same for more advanced static website, e.g. based on popular static site generators.

Reviewer 2: ```suggestion
The purpose of this example is to show how to deploy a static website in an asset canister. While the website in this example is very simple, the method would be the same for a more advanced static website, e.g. based on popular static site generators.
```

This example covers:

- Build a very simple HTML website
- Create the dfx.json file
- Deploy the canister smart contract locally
- Test the frontend in browser 

## Installation
This example project can be cloned, installed and deployed locally, for learning and testing purposes. The instructions are based on running the example on either macOS or Linux, but when using WSL2 on Windows, the instructions will be the same.

### Prerequisites

> The example project requeres the following installed:

Reviewer 2: ```suggestion
The example project requires the following installed:
```

> - GitHub

Reviewer 2: ```suggestion
- git
```

- DFX 

> GitHub can be installed from various package managers. DFX can be installed following the instructions [here](https://smartcontracts.org/docs/quickstart/local-quickstart.html#download-and-install).

Reviewer 2: ```suggestion
git can be installed from various package managers. DFX can be installed following the instructions [here](https://smartcontracts.org/docs/quickstart/local-quickstart.html#download-and-install).
```

### Install
Install the example dapp project:

```bash
$ git clone https://github.com/dfinity/examples
$ cd motoko/static-website
```

## Documentation

> There are two parts in this dapp, one is the website and the other is the `dfx.json`. Since there's no backend, there is not any benefit of using the `dfx new project_name` to setup a template, the `dfx.json` file is all there is needed.

Reviewer 2: (corrected run-on sentences and grammar)
```suggestion
There are two parts in this dapp. One is the website and the other is the `dfx.json`. Since there's no backend, there is not any benefit of using the `dfx new project_name` to setup a template. The `dfx.json` file is all that is needed.
```

### Website

> he website is really simple, it consists of a HTML file, CSS file and a PNG file. The content of the HTML file looks like this:

Reviewer 2: ```suggestion
The website is really simple. It consists of a HTML file, a CSS file and a PNG file. The content of the HTML file looks like this:
```

```html
<!doctype html>
<html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width">
        <title>Static Website</title>
        <base href="/">
        <link type="text/css" rel="stylesheet" href="styles.css" />
    </head>
    <body>
        <img src="logo.png" alt="DFINITY logo" />
    </body>
</html>
```

The only styling done in the CSS file is aligning the logo image:

```css
img {
    max-width: 50vw;
    max-height: 25vw;
    display: block;
    margin: auto;
}
```

> The top level folder is called `assets` and inside this folder, the CSS file and image is stored in a folder called `assets` and the HTML file `index.html` is stored in a folder called `src`. 

Reviewer 2: For the purposes of this simple example, since these three files end up deploying into the same place, would it make sense to move them into a single directory?  This would avoid having an `assets/assets/` directory.
```
hosting/
    static-website/
        assets/
            index.html
            main.css
            logo.png
```

### dfx.json
The `dfx.json` file is a configuration file which specifies the canister used for the dapp. In this case only one canister is needed, and besides the canister configuration, `dfx.json` also includes information about DFX version, build settings and network settings.

```json
{
    "canisters": {
        "www": {
            "frontend": {
                "entrypoint": "assets/src/index.html"
            },
            "source": [
                "assets/assets",
                "assets/src"
            ],
            "type": "assets"
        }
    },
    "defaults": {
        "build": {
            "args": "",
            "packtool": ""
        }
    },
    "dfx": "0.8.4",
    "networks": {
        "local": {
            "bind": "127.0.0.1:8000",
            "type": "ephemeral"
        }
    },
    "version": 1
}
```

This is all needed for creating a canister smart contract for hosting a static website on the IC.

## Deployment
The local network is started by running this command:

```bash
$ dfx start --background
```

When the local network is up and running, run this command to deploy the canisters:

```bash
$ dfx deploy
```

## Testing

> The URL for the frontend is depending on the canister ID, the local canister IDs can be found in `.dfx/local/canister_ids.json`. When deployed, the URL will look like this:

Reviewer 2: ```suggestion
The URL for the frontend depends on the canister ID.  The local canister IDs can be obtained using `dfx canister id`, in this case `dfx canister id www`. When deployed, the URL will look like this:
```

> **https://<ui_canister_id>.localhost:8000**

Reviewer 2: I wonder if we should note: this works for Chrome, Firefox, Opera, and I'm sure more.  Unfortunately it doesn't for Safari, which would require a URL like `http://localhost:8000?canisterId=<ui_canister_id>`

Also, for local development, only `http://` will work, not `https://`
```suggestion
**http://<ui_canister_id>.localhost:8000**
```

![Website](README_images/website.png)

## Introduction
The purpose of this example is to show how to deploy a static website in an asset canister. While the website in this example is very simple, the method would be the same for a more advanced static website, e.g. based on popular static site generators.

This example covers:

- Build a very simple HTML website
- Create the dfx.json file
- Deploy the canister smart contract locally
- Test the frontend in browser 

## Installation
This example project can be cloned, installed and deployed locally, for learning and testing purposes. The instructions are based on running the example on either macOS or Linux, but when using WSL2 on Windows, the instructions will be the same.

### Prerequisites
The example project requires the following installed:

- git
- dfx 

git can be installed from various package managers. DFX can be installed following the instructions [here](https://smartcontracts.org/docs/quickstart/local-quickstart.html#download-and-install).

### Install
Install the example dapp project:

```bash
$ git clone https://github.com/dfinity/examples

> $ cd motoko/static-website

Reviewer 2: ```suggestion
$ cd examples/hosting/static-website
```
